libr/bin/format/mach0/mach0.c

```diff
@@ -3227,44 +3220,61 @@ RSkipList *MACH0_(get_relocs)(struct MACH0_(obj_t) *bin) {
 			return NULL;
 		}
 
-		for (ur.p = opcodes, end = opcodes + amount ; ur.p < end; ) {
-			ut8 imm = *ur.p & BIND_IMMEDIATE_MASK, op = *ur.p & BIND_OPCODE_MASK;
-			++ur.p;
-			if (ur.p >= end) {
-				break;
-			}
-			switch (op) {
+		int partition_sizes[] = {bind_size, lazy_size, weak_size}, partition_idx;
+		int opcodes_offset = 0;
+		for (partition_idx = 0; partition_idx < R_ARRAY_SIZE (partition_sizes); partition_idx++) {
+			int partition_size = partition_sizes[partition_idx];
+
+			ut8 type = 0;
+			int lib_ord = 0, seg_idx = -1, sym_ord = -1;
+			char *sym_name = NULL;
+			size_t j, count, skip;
+			st64 addend = 0;
+			ut64 addr = bin->segs[0].vmaddr;
+			ut64 segment_end_addr = addr + bin->segs[0].vmsize;
+
+			ulebr ur = {opcodes + opcodes_offset};
+			ut8 *end = ur.p + partition_size;
+			bool done = false;
+			while (!done && ur.p < end) {
+				ut8 imm = *ur.p & BIND_IMMEDIATE_MASK, op = *ur.p & BIND_OPCODE_MASK;
+				++ur.p;
+				switch (op) {
 #define ULEB() read_uleb128 (&ur,end)
 #define SLEB() read_sleb128 (&ur,end)
-			case BIND_OPCODE_DONE:
-				break;
-			case BIND_OPCODE_THREADED: {
-				switch (imm) {
-				case BIND_SUBOPCODE_THREADED_SET_BIND_ORDINAL_TABLE_SIZE_ULEB: {
-					ut64 table_size = ULEB();
-					if (table_size == 0 || table_size > 65535) {
-						bprintf ("Error: BIND_SUBOPCODE_THREADED_SET_BIND_ORDINAL_TABLE_SIZE_ULEB size is wrong\n");
-						break;
-					}
-					if (threaded_binds) {
-						r_pvector_free (threaded_binds);
-					}
-					threaded_binds = r_pvector_new ((RPVectorFree) &free);
-					if (threaded_binds) {
-						r_pvector_reserve (threaded_binds, table_size);
-						threaded_binds->v.len = table_size;
-						sym_ord = 0;
+				case BIND_OPCODE_DONE: {
+					bool in_lazy_binds = partition_idx == 1;
+					if (!in_lazy_binds) {
+						done = true;
 					}
 					break;
 				}
-				case BIND_SUBOPCODE_THREADED_APPLY:
-					if (threaded_binds) {
-						if (addr && seg_idx >= 0) {
-							ut64 cursor = addr;
+				case BIND_OPCODE_THREADED: {
+					switch (imm) {
+					case BIND_SUBOPCODE_THREADED_SET_BIND_ORDINAL_TABLE_SIZE_ULEB: {
+						ut64 table_size = ULEB();
+						if (table_size == 0 || table_size > 65535) {
```

Comment:
use constants or move this logic into a separate function to make the code more readable (`if (is_valid (table_size)) {`). Also use UT16_MAX instead of 65535